core/common/src/test/scala/net/liftweb/common/BoxSpec.scala

```diff
@@ -141,6 +141,15 @@ class BoxSpec extends Specification with ScalaCheck with BoxGenerator {
     "define a 'flatMap' method transforming its value in another Box. If the value is transformed in an Empty can, the total result is an Empty can" in {
       Full(0) flatMap { x: Int => if (x > 0) Full("full") else Empty } must beEmpty
     }
+    "define a 'flatten' method if it contains another Box. If the inner box is a Full can, the final result is a Full can" in {
+      Full(Full(1)).flatten must_== Full(1)
+    }
+    "define a 'flatten' method if it contains another Box. If the inner box is a Failure, the final result is a Failure" in {
+      Full(Failure("error", Empty, Empty)).flatten must_== Failure("error", Empty, Empty)
```

Comment:
This is an interesting question. This seems like it diverges from typical Scala collection library behavior; in particular, classic `flatten` behavior preserves the *outer* type, not the inner one:

```scala
@ Vector(List(1,2,3)).flatten
res2: Vector[Int] = Vector(1, 2, 3)
@ Vector(List()).flatten
res3: Vector[Nothing] = Vector()
@ 
```

I think what we want to do is make sure we behave the same as an `identity`-based `flatMap`:

```scala
@ Full(Failure("bam")).flatMap(identity)
res11: Box[Nothing] = Failure(bam,Empty,Empty)
```

So this is indeed the correct behavior based on how we've defined `flatMap`. Whether or not that's “correct”, I think changing the behavior this late in the game would be waay too problematic, so we'll leave it as is for now.